DB_HW_Recruitmentagency

h6bi opened the pull request:
Sorry, accidentally pushed to main first try, hence the reverts.

h6bi committed:
Revert "Create DB_Abel_Horvath_HW_Recruitmentagency_descriptions.docx"

This reverts commit 1a0a25b45dccc7f942fcea4e08aa9d515ce3b366.

h6bi committed:
Revert "Create DB_Abel_Horvath_HW_Recruitmentagency_diagram.png"

This reverts commit 7c8a372d2211dbf4d0f149f4b747c9c6dc5f6993.

h6bi committed:
Create DB_Abel_Horvath_HW_Recruitmentagency_diagram.png

h6bi committed:
Create DB_Abel_Horvath_HW_Recruitmentagency_descriptions.docx

KseniyaLazarchik reviewed (commented):
Hi @h6bi 
Well done
Some issues

1. candidates: country_of_residence, city_of_residence
Currently, country and city are stored as varchar(50) inside candidates. And in job_listings: country, city (similar problem)
This can lead to data duplication (for example, "USA" can be written as "United States", "U.S.", "USA").

You can create new cities and cities tables and link job_listings (candidates) to the new tables via id.

2. companies: registered_office (address in varchar(100))
Currently, the entire address string is stored in one field, which makes searching by city (country?) and street impossible.

You can move the address to a new table, take the city (country?) as in the example above.

3.candidates.candidate_email (+interviewer) as PK is a bad idea
Email is unstable, it can be changed, which will create problems in FK.
It is better to use candidate_id INT AUTO_INCREMENT and make candidate_email UNIQUE.

4. CHAR(8) instead of INT (you can add a surrogate key) for keys (job_ID, company_ID, trainer_ID)
Using CHAR(8) for identifiers is not optimal.
INT is much faster in indexes and JOIN.

5. candidate_experience company name is better to take from the company table

6. Experience and Skills are better to be split into 2 different tables and then merge them with Candidate and Job via bridge tables (expreience_req and skill_req, candidate_experience and candidate_skills)

h6bi committed:
added week 2 HW

h6bi committed:
Week 1 HW correction

1. added location table
2. added company_address table
3. replaced email PKs with IDs
4. changed PK data types as integers
5. rewokred skills and experience tables and linked them with jobs and candidates via bridge tables

h6bi commented:
Thank you for your feedback! Commited the correction for your review.

h6bi committed:
correct files

Messed up the commit the first try (again), sorry.

KseniyaLazarchik reviewed (commented):
Hi @h6bi 

Good
Move job_role to a separate table (as discussed during screening)
Also here is my proposal for implementing geo hierarchy
![image](https://github.com/user-attachments/assets/0b79765a-f3a9-4c4b-a5e5-3f4845cb442d)

h6bi committed:
WEEK 2 HW correction nr 1

Hi!

Made the requested corrections, and uploaded it in one file.

I have one question about your review:
In part 1.2 what do you mean by "IFNULL is used to handle NULL.". As far is I know in Postgresql COALESCE is used instead of IFNULL.

Thanks! :)

h6bi committed:
WEEK 1 HW correction nr 2

- added seperate role table and implemented geo hierarchy as suggested

- made some minor reworks (e.g. interviewer_ round column name instead of round) for week 4 task

h6bi commented:
Thank you for your feedback! :) Commited these corrections:

- added seperate role table and implemented geo hierarchy as suggested

- made some minor reworks (e.g. interviewer_ round column name instead of round) for week 4 task

KseniyaLazarchik reviewed (approved):
Hi @h6bi 
Good job
You can merge

h6bi committed:
Merge pull request #3 from h6bi/SQL_SELECT_HW

SQL_SELECT_HW
